pallets/loans-ref/src/migrations.rs

```diff
@@ -0,0 +1,87 @@
+use frame_support::{
+	ensure,
+	pallet_prelude::ValueQuery,
+	storage, storage_alias,
+	traits::{Get, OnRuntimeUpgrade},
+	weights::Weight,
+	Blake2_128Concat,
+};
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+use crate::*;
+
+mod v0 {
+	use super::*;
+
+	/// This storage comes from the previous pallet loans.
+	/// It is used as an indicator that the previous pallet loans still exists.
+	/// If this storage is not found, the nuking process is aborted.
+	#[storage_alias]
+	pub(crate) type NextLoanId<T: Config> =
+		StorageMap<Pallet<T>, Blake2_128Concat, PoolIdOf<T>, u128, ValueQuery>;
+}
+
+/// This migration nukes all storages from the pallet individually.
+pub struct NukeMigration<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for NukeMigration<T> {
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+		ensure!(
+			contains_prefixed_key(&loan_prefix()),
+			"Pallet loans prefix doesn't exists"
+		);
+
+		ensure!(
+			v0::NextLoanId::<T>::iter_values().count() == 1,
+			"Pallet loans contains doesn't contain old data"
+		);
+
+		Ok(Vec::new())
+	}
+
+	fn on_runtime_upgrade() -> Weight {
+		if v0::NextLoanId::<T>::iter_values().count() == 1 {
```

Comment:
Ideally, I would have wanted to check here if the `on_chain_storage_version()` was 0, and bump the version to 1, but I could not do it because if I initialize the pallet with `STORAGE_VERSION = 1`, then it never enters here, because it will be already 1.

I think the issue was not setting the version to 0 in the past.

Reply:
> I think the issue was not setting the version to 0 in the past.

This makes me think that maybe we should add this version for all pallets, in this way, if in the future we have a new migration, we can easily fetch the correct version number 0 for them.

Reply:
Yes, I have been meaning to open an issue for that. [Here it is](https://github.com/centrifuge/centrifuge-chain/issues/1295) 🌚

Reply:
Great! ❤️